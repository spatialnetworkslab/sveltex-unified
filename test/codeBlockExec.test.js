
import dedent from 'dedent'
import { processor } from '../src/index.js'

const process = async (md) => {
  const result = await processor()
    .process(md)
  return result.toString()
}

test('process js exec code block and normal script block', async () => {
  const input = await process(dedent`
        <script>
            let d = 'e'
        </script>
        \`\`\`js exec
        let a = 'b'
        let c = 1
        \`\`\`
        
        { a }
    `)
  expect(input).toMatchSnapshot()
})

test('process js exec echo code block', async () => {
  const input = await process(dedent`
          \`\`\`js exec echo
          let a = 'b'
          let c = 1
          \`\`\`
          
          { a }
      `)
  expect(input).toMatchSnapshot()
})

test('process multiple js exec code blocks', async () => {
  const input = await process(dedent`
            \`\`\`js exec
            let a = 'b'
            let c = 1
            \`\`\`
            
            { a }

            \`\`\`js exec
            let d = 'b'
            let e = 1
            \`\`\`
        `)
  expect(input).toMatchSnapshot()
})

test('process js module code block', async () => {
  const input = await process(dedent`
            <script context="module">
                let d = 'e'
            </script>
            \`\`\`js module
            let x = 'b'
            let y = 1
            \`\`\`
            
            { a }
        `)
  expect(input).toMatchSnapshot()
})

test('process html exec echo code block', async () => {
  const input = await process(dedent`
          \`\`\`html exec echo
          <p>test</p>
          \`\`\`
      `)
  expect(input).toMatchSnapshot()
})
