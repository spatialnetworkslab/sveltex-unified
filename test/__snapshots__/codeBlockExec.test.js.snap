// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`process html exec echo code block 1`] = `
"<pre><code class=\\"echo_html\\">&#60;p&#62;test&#60;/p&#62;
</code></pre>
<div class=\\"exec_html\\"><p>test</p></div>"
`;

exports[`process js exec code block and normal script block 1`] = `
"<script>

    let d = 'e'

let a = 'b'
let c = 1
</script>

<p>{ a }</p>"
`;

exports[`process js exec echo code block 1`] = `
"<script>
let a = 'b'
let c = 1
</script>

<pre class=\\"language-js\\"><code class=\\"language-js\\"><span class=\\"token keyword\\">let</span> a <span class=\\"token operator\\">=</span> <span class=\\"token string\\">&#39;b&#39;</span>
<span class=\\"token keyword\\">let</span> c <span class=\\"token operator\\">=</span> <span class=\\"token number\\">1</span>
</code></pre>
<p>{ a }</p>"
`;

exports[`process js module code block 1`] = `
"<script context=\\"module\\">

    let d = 'e'

let x = 'b'
let y = 1
</script>

<p>{ a }</p>"
`;

exports[`process multiple js exec code blocks 1`] = `
"<script>
let a = 'b'
let c = 1
let d = 'b'
let e = 1
</script>

<p>{ a }</p>"
`;
